static/src/javascripts/projects/commercial/modules/dfp/Advert.ts

```diff
@@ -84,21 +121,28 @@ class Advert {
 		adSlotNode: HTMLElement,
 		additionalSizeMapping: SizeMapping = {},
 	) {
+		// Try to used size mappings if available
 		const defaultSizeMappingForSlot = adSlotNode.dataset.name
 			? getSlotSizeMapping(adSlotNode.dataset.name)
 			: {};
 
-		const sizeMapping = mergeSizeMappings(
+		let sizeMapping = mergeSizeMappings(
 			defaultSizeMappingForSlot,
 			additionalSizeMapping,
 		);
 
+		// If the size mapping is empty, use the data attributes to create a size mapping, this is used on some interactives
 		if (isSizeMappingEmpty(sizeMapping)) {
-			throw new Error(
-				`Tried to render ad slot '${
-					adSlotNode.dataset.name ?? ''
-				}' without any size mappings`,
-			);
+			sizeMapping = getAdBreakpointSizes(adSlotNode);
```

Comment:
Could we name this something like `getBreakpointSizesFromSlotAttributes`

Or whatever you think is a better name that indicates its coming from the slot data attributes?